src/DynamoCoreWpf/Views/Core/DynamoOpenFileDialog.cs

```diff
@@ -120,14 +121,11 @@ public string SelectedPath
         {
             get
             {
-                string selectedPath;
-                IShellItem item;
-                dialog.GetResult(out item);
-                item.GetDisplayName(SIGDN.SIGDN_FILESYSPATH, out selectedPath);
                 return selectedPath;
             }
             set
             {
+                selectedPath = value;
```

Comment:
What if the setter method is called after the dialog is disposed? Better to keep this property independent of dialog and have a specific method to set the file path in dialog, that take dialog as input rather than using the member variable. In fact, we shouldn't cache dialog object as we don't guarantee the life of dialog as same as this object.

public string SelectedPath {get; private set;}